src/cli/rustup_mode.rs

```diff
@@ -55,17 +55,46 @@ where
 pub fn main() -> Result<utils::ExitCode> {
     self_update::cleanup_self_updater()?;
 
+    use clap::ErrorKind::*;
     let matches = match cli().get_matches_from_safe(process().args_os()) {
         Ok(matches) => Ok(matches),
-        Err(e)
-            if e.kind == clap::ErrorKind::HelpDisplayed
-                || e.kind == clap::ErrorKind::VersionDisplayed =>
-        {
-            writeln!(process().stdout().lock(), "{}", e.message)?;
+        Err(clap::Error {
+            kind: HelpDisplayed,
+            message,
+            ..
+        }) => {
+            writeln!(process().stdout().lock(), "{}", message)?;
+            return Ok(utils::ExitCode(0));
+        }
+        Err(clap::Error {
+            kind: VersionDisplayed,
+            message,
+            ..
+        }) => {
+            writeln!(process().stdout().lock(), "{}", message)?;
+            info!("This is the version for the rustup toolchain manager, not the rustc compiler.");
```

Comment:
I hope using the log macros is fine; not sure what the `process().stdout().lock()` dance does here.

Reply:
https://github.com/rust-lang/rustup/blob/master/CONTRIBUTING.md#no-direct-use-of-process-state-outside-rustupcurrentprocess

info! does the dance for you, as long as its info! from our macros, not a new import or some such.

Reply:
Yep, it's the already imported rustup macro. Thanks for the info! (pun intended)